modules/xml/src/main/java/org/mule/module/xml/xpath/JaxpXPathEvaluator.java

```diff
@@ -64,20 +60,15 @@
 public abstract class JaxpXPathEvaluator implements XPathEvaluator, XPathVariableResolver
 {
 
+    private static final int MIN_IDLE_XPATH_EXPRESSIONS = 1;
+    private static final int MAX_IDLE_XPATH_EXPRESSIONS = 32;
+    private static final int MAX_ACTIVE_XPATH_EXPRESSIONS = MAX_IDLE_XPATH_EXPRESSIONS;
+
     private final XPathFactory xpathFactory;
     private final Map<String, String> prefixToNamespaceMap = new HashMap<>();
     private final ThreadLocal<MuleEvent> evaluationEvent = new ThreadLocal<>();
+    private Map<String, GenericObjectPool<XPathExpression>> xpathExpressionPools = new HashMap<>();
```

Comment:
keep using the loading cache here, since it better handles concurrency, so you can avoid having to synchronixe on this map